opentracing-metrics-micrometer/src/main/java/io/opentracing/contrib/metrics/micrometer/MicrometerMetricsReporter.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2017-2018 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.metrics.micrometer;
+
+import io.micrometer.core.instrument.Metrics;
+import io.opentracing.contrib.api.SpanData;
+import io.opentracing.contrib.metrics.AbstractMetricsReporter;
+import io.opentracing.contrib.metrics.MetricLabel;
+import io.opentracing.contrib.metrics.MetricsReporter;
+import io.opentracing.contrib.metrics.label.BaggageMetricLabel;
+import io.opentracing.contrib.metrics.label.ConstMetricLabel;
+import io.opentracing.contrib.metrics.label.TagMetricLabel;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This class provides a Micrometer based implementation of the {@link MetricsReporter}.
+ *
+ */
+public class MicrometerMetricsReporter extends AbstractMetricsReporter implements MetricsReporter {
+    private String name;
+
+    protected MicrometerMetricsReporter(String name, List<MetricLabel> labels) {
+        super(labels);
+        this.name = name;
+    }
+
+    @Override
+    public void reportSpan(SpanData spanData) {
+        List<String> tags = new ArrayList<>(this.metricLabels.length*2);
+        for (MetricLabel metricLabel : this.metricLabels) {
+            tags.add(metricLabel.name());
+            tags.add(metricLabel.value(spanData).toString());
```

Comment:
Need to guard against a `null` value being returned. This is also used as an indication that the metric shouldn't be reported (currently) - we may want to find a better approach in the future, but for now best to be consistent with the prometheus only approach.

So if a `null` value is returned, break out of the loop and don't record the timer.